src/bio.c

```diff
@@ -250,19 +241,8 @@ void *bioProcessBackgroundJobs(void *arg) {
     bio_worker_num = bioWorkerNum(bwd);
 
     while (1) {
-        listNode *ln;
-
-        /* The loop always starts with the lock hold. */
-        if (listLength(bwd->bio_jobs) == 0) {
-            pthread_cond_wait(&bwd->bio_newjob_cond, &bwd->bio_mutex);
-            continue;
-        }
-        /* Get the job from the queue. */
-        ln = listFirst(bwd->bio_jobs);
-        job = ln->value;
-        /* It is now possible to unlock the background system as we know have
-         * a stand alone job structure to process.*/
-        pthread_mutex_unlock(&bwd->bio_mutex);
+        /* Get job - blocking until available */
+        job = mutexQueuePop(bwd->bio_jobs, true);
```

Comment:
```suggestion
        bio_job *job = mutexQueuePop(bwd->bio_jobs, true);
```

Reduces scope of `job` and removes one more line above.